tools/codegen/gen.py

```diff
@@ -1293,6 +1375,18 @@ def key_func(fn: NativeFunction) -> str:
 
     cpu_fm.write('Functions.cpp', lambda: {})
 
+    if options.enable_codegen_unboxing:
```

Comment:
It sounds like the idea of the current PR is that:
- we add this flag to optionally generate unboxing codegen, and only use it for mobile
- torchscript (server-side) won't use the unboxing codegen at all (but it would in a later PR? And then the unboxing codegen part would get generated all of the time, so we wouldn't want the flag?)

The main risk that I see is that the unboxing codegen lands without getting tested outside of mobile. So I think a separate PR is fine if it's easier to split up that way, but it would be good to land the changes all at once, including  usage in torchscript.